### DISCUSSION TITLE
Why minify features like Cloudflare autominify can be problematic for VuePress

### DISCUSSION THREAD

**Discussion by Ackerven** [vuepress-theme-hope/vuepress-theme-hope#3752] [Q&A] [answered]
Due to some reasons, accessing the github.io page in China may be inaccessible. In this case, it is necessary to use Cloudflare acceleration. However, when using Cloudflare, turning on some acceleration options will cause duplication of the homepage.

![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/70303900/b2c5feb4-284e-48d5-84f8-8fb7b0c9cbe1)

---

Steps to reproduce this problem:
1. Create a Vuepress-Theme-Hope project of `docs` type

    ```sh
      ~/tmp$ pnpm create vuepress-theme-hope minirepo
      .../share/pnpm/store/v3/tmp/dlx-80757    |  +83 ++++++++
      .../share/pnpm/store/v3/tmp/dlx-80757    | Progress: resolved 83, reused 83, downloaded 0, added 83, done
      ? Select a language to display / 选择显示语言 english (US)
      ? Choose package manager pnpm
      Generating package.json...
      ? Your project name vuepress-theme-hope-template
      ? Your project version 2.0.0
      ? Your project description A project of vuepress-theme-hope
      ? Your project license MIT
      ? Does the project need multiple languages? No
      ? Do you need a GitHub workflow to deploy docs on GitHub pages? Yes
      ? What type of project do you want to create? docs
      Generating Template...
      ? Initialize a git repository? Yes
      Installing Deps...
      This may take a few minutes, please be patient.
    ```

3. Open the Github Pages in the repository and add the domain `example.com`
5. Host the domain `example.com` to Cloudflare, add a CNAME record, point `example.com` to `xxx.github.io`, and turn on **Proxy status**

    ![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/70303900/2f084e92-658e-4a5c-b4a3-2fd70b7890fc)

7. Check **JavaScript**, **CSS**, **HTML** in **Speed -> Optimization -> Content Optimization -> Auto Minify**

    ![image](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/70303900/ee2f69e3-afc0-4199-be0c-6568c22246b8)

9. Use `example.com` to access

---

**I know Cloudflare has alerted me to possible incompatibilities, is this possible to be fixed?**

---

Minimal reproduction:

+ repo: [theme-hope-minirepo](https://github.com/Ackerven/theme-hope-minirepo).
+ website: [https://vth.fne.cc](https://vth.fne.cc)

**Comment by Ackerven** [comment]
By the way, when adding a domain to Cloudflare, Cloudflare will display a wizard to guide users to enable some acceleration options.

**Comment by Mister-Hope** [comment, marked as answer]
- https://theme-hope.vuejs.press/faq/common-error.html#hydration-completed-but-contains-mismatches 

I believed needed information is already added at the FAQ part, and it "clearly mentions how the issue happens: Vue Hydration is sensitive with white space“.

Generally speaking:

Though the following html render the same result:

```html
<p>VuePress Theme Hope is powerful</p>

<p>VuePress    Theme
Hope is powerful</p>
```

The innerText is different, and it will not be the same for the following Vue component:

```vue
<template>
  <p v-text="text" /> 
</template>
<script setup>
import { ref } from 'vue';

const text = ref('VuePress Theme Hope is powerful')
// const text = ref('VuePress    Theme\nHope is powerful')
```

So the auto minify MUST be off.

Further notes:

The `@vue/server-render` always trim white space if possible, so if you use v-for for the above component, then the result will be:

```html
<p>xxx</p><p>xxx</p><p>xxx</p>
         👆 no white space here
```

Any white space between tags will be trimmed already, so the HTML file DOES not need minify at all, all preserved white space is probably connected with the component data.

E.g.: view-source:https://theme-hope.vuejs.press/faq/common-error.html#you-are-not-allowed-to-use-plugin-xxx-yourself-in-vuepress-config-file

The source file is only wrapped in `<pre>` elements

---

That's enough for my explanation. If you want further discussion or details, ahead to places like Vue Forum or vue/core discussion panel. This is related to "Vue SSR Hydration" and nothing with VuePress and my theme.

**Reply by Ackerven** [reply]
Thanks!

**Reply by Mister-Hope** [reply]
BTW, I believe css and js minify can be on, can you check that?

**Reply by Ackerven** [reply]
You're right, it works fine.

**Comment by Deflaimun** [comment]
Looks like the feature is now deprecated 
https://community.cloudflare.com/t/deprecating-auto-minify/655677/7

So the FAQ no longer applies.
